src/Signature/S3SignatureV4.php

```diff
@@ -41,6 +41,11 @@ public function presign(
                 'X-Amz-Content-Sha256',
                 $this->getPresignedPayload($request)
             );
+        } else {
+            $request = $request->withHeader(
```

Comment:
This is a no-op. You're assigning the current value of the header to the be the new value of the header.